zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/ZkClientUriDomainMappingHelper.java

```diff
@@ -181,4 +154,37 @@ public void updateDomainBasedAuthInfo(ServerCnxn cnxn) {
       }
     }
   }
+
+  /**
+   * The watcher used to listen on client uri - domain mapping root path for mapping update
+   * Extends DumbWatcher instead of ServerCnxn here some methods in ServerCnxn are not accessible from here
```

Comment:
Technically the abstract methods are not a problem, because we can leave the implementations empty, as how DumbWatcher class is implemented. The only problem is there some APIs are with package-private modifier, but the place where our class exists is in a different package with ServerCnxn, so we can not access them here. For example  `abstract void setSessionId(long sessionId);`